pylint/extensions/magic_value.py

```diff
@@ -37,11 +38,30 @@ class MagicValueChecker(BaseChecker):
                 "default": (0, -1, 1, "", "__main__"),
                 "type": "csv",
                 "metavar": "<argument names>",
-                "help": " List of valid magic values that `magic-value-compare` will not detect.",
+                "help": " List of valid magic values that `magic-value-compare` will not detect,"
+                "Supports integers, floats, negative numbers, for empty string enter ``''``,"
+                " for backslash values just use one backslash e.g \\n.",
             },
         ),
     )
 
+    def __init__(self, linter: PyLinter) -> None:
+        """Initialize checker instance."""
+        super().__init__(linter=linter)
+        self.valid_magic_vals: tuple[(float | str), ...] = ()
+
+    def open(self) -> None:
+        if self._is_default_magic_vals():
```

Comment:
Let's rename this to something like `_is_split_magic_vals_option` cause that is what we are essentially checking.

Reply:
we are checking if the valid-magic-values config is taken from default (being a tuple) or from the rcfile (not being a tuple)

Reply:
maybe we can rename it to: `_is_magic_vals_externally_parsed`

Reply:
`_is_magic_vals_set_by_user`?